market_maker_keeper/leverjfutures_market_maker_keeper_v2.py

```diff
@@ -0,0 +1,331 @@
+                quote_allocation = total_available_wad/number_of_partitions_for_allocation
+                self.logger.debug(f'open_position_for_base: {open_position_for_base}')
+                # bids are made basis quote_allocation and asks basis base_allocation
+                # if open position is net long then quote_allocation is adjusted.
+                # if open position is net too long then target_price is adjusted to reduce price of the asks/offers
+                if (open_position_for_base.value > 0):
+                    open_position_for_base_in_quote = open_position_for_base*average_price
+                    net_adjusted_quote_value = quote_allocation.value - abs(open_position_for_base_in_quote.value)
+                    self.logger.debug(f'net_adjusted_quote_value: {net_adjusted_quote_value}')
+                    quote_allocation = Wad(net_adjusted_quote_value) if net_adjusted_quote_value > minimum_required_balance.value else Wad(0)
+                    # if open position is within 1 Wad range or more than quote allocations then target price is leaned down by 0.1 percent
+                    if Wad(net_adjusted_quote_value) < Wad(1):
+                        self.target_price_lean = Wad.from_number(0.999)
+                    else:
+                        self.target_price_lean = Wad(0)
+                elif (open_position_for_base.value < 0):
+                    # if open position is net short then base_allocation is adjusted
+                    # if open position is net too short then target_price is adjusted to increase price of the bids
+                    net_adjusted_base_value = base_allocation.value - abs(open_position_for_base.value)
+                    minimum_required_balance_in_base = minimum_required_balance/average_price
+                    self.logger.debug(f'net_adjusted_base_value: {net_adjusted_base_value}')
+                    base_allocation = Wad(net_adjusted_base_value) if net_adjusted_base_value > minimum_required_balance_in_base.value else Wad(0)
+                    # if open position is within 1 Wad range or more than base allocations then target price is leaned up by 0.1 percent
+                    if Wad(net_adjusted_base_value) < Wad(1):
+                        self.target_price_lean = Wad.from_number(1.001)
+                    else:
+                        self.target_price_lean = Wad(0)
+        else:
+            base_allocation = Wad(0)
+            quote_allocation = Wad(0)
+
+        allocation = {base: base_allocation, quote: quote_allocation}
+        self.logger.debug(f'allocation: {allocation}')
+        return allocation
+
+    def our_sell_orders(self, our_orders: list) -> list:
+        return list(filter(lambda order: order.is_sell, our_orders))
+
+    def our_buy_orders(self, our_orders: list) -> list:
+        return list(filter(lambda order: not order.is_sell, our_orders))
+
+    def adjust_target_price(self, target_price):
```

Comment:
Would be helpful to have typings to differentiate `Wad` or `Price` objects

Reply:
Also, `assert (isinstance(target_price, Wad))` at the start to catch any argument errors, which simplifys debugging

Reply:
yup, will add typings to this